File: code/done/242.valid-anagram.go
/*
 * @lc app=leetcode id=242 lang=golang
 *
 * [242] Valid Anagram
 */

// @lc code=start
func isAnagram(s string, t string) bool {
	record := [26]int{}
	for _, r := range s {
		record[r-rune('a')]++
	}
	for _, r := range t {
		record[r-rune('a')]--
	}

	return record == [26]int{}
}

// @lc code=end

File: code/done/242.valid-anagram.py
#
# @lc app=leetcode id=242 lang=python3
#
# [242] Valid Anagram
#


# @lc code=start
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        d = [0]*26
        for ch in s:
            d[ord(ch)-ord('a')] += 1
        for ch in t:
            d[ord(ch)-ord('a')] -= 1
        return d == [0]*26


# @lc code=end
